src/e3/encoding/vex.py

```diff
@@ -0,0 +1,902 @@
+        }
+
+
+class AuthorRole(Enum):
+    """Role of the document author.
+
+    The author role **MAY** specify the role of the *author*.
+
+    The author role **MAY** use the *category of publisher* roles defined by
+    CSAF 2.0:
+
+    :cvar COORDINATOR: indicates individuals or organizations that manage a
+        single vendor’s response or multiple vendors’ responses to a
+        vulnerability, a security flaw, or an incident. This includes all
+        Computer Emergency/Incident Response Teams (CERTs/CIRTs) or agents
+        acting on the behalf of a researcher.
+    :cvar DISCOVERER: indicates individuals or organizations that find
+        vulnerabilities or security weaknesses. This includes all manner of
+        researchers.
+    :cvar TRANSLATOR: indicates individuals or organizations that translate
+        VEX documents. This includes all manner of language translators, also
+        those who work for the party issuing the original advisory.
+    :cvar OTHER: indicates a catchall for everyone else. Currently, this
+        includes editors, reviewers, forwarders, republishers, and miscellaneous
+        contributors.
+    :cvar USER: indicates anyone using a vendor’s product.
+    :cvar VENDOR: indicates developers or maintainers of information system
+        products or services. This includes all authoritative product vendors,
+        Product Security Incident Response Teams (PSIRTs), and product resellers
+        and distributors, including authoritative vendor partners.
+    """
+
+    COORDINATOR = "coordinator"
+    DISCOVERER = "discoverer"
+    OTHER = "other"
+    TRANSLATOR = "translator"
+    USER = "user"
+    VENDOR = "vendor"
+
+    @classmethod
+    def from_value(
+        cls, value: str | AuthorRole | None, default: AuthorRole | str = OTHER
+    ) -> AuthorRole:
+        if isinstance(value, str) and value:
+            return cls(value)
+        elif isinstance(value, cls):
+            return value
+        elif default:
+            # The default value is either an AuthorRole or a string, better
+            # return a AuthorRole.
+            return cls.from_value(default)
+        else:
+            raise ValueError(f"Invalid default value {default}")
+
+
+class Document(JsonData):
+    """Vulnerabilities Exploitability Exchange document.
+
+    A VEX document is a binding of product information, vulnerability
+    information, and the relevant status details relating them.
+
+    Minimum data elements of a VEX document must include the VEX metadata,
+    product details, vulnerability details, and product status.
+
+    :ivar Metadata metadata: VEX metadata.
+    :ivar ProductStatus status: Status information about a vulnerability in that
+        product.
+    :ivar list[Statement] | None statements: The list of statements defined for
+        this VEX document.
+    """
+
+    def __init__(
+        self,
+        metadata: Metadata,
+        statements: list[Statement] | None = None,
+    ):
```

Comment:
missing docstring

Reply:
After adding all the docstrings to `__init__()` methods, I saw that the doc build does not include them :)